In [14]:
# Import packages

import streamlit as st
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import altair

Write code to start Streamlit web app

st.write("Hello world")

Figured out that Streamlit requires a .py (python) file, not .ipynb (notebook).

In [15]:
# Launched Streamlit with Employee.csv using command "Streamlit run app.py"

pr_df = pd.read_csv("PerformanceRating.csv")
st.write(pr_df)


Checked for null values using df.info(). There are 0 missing values.

In [16]:
pr_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6709 entries, 0 to 6708
Data columns (total 11 columns):
 #   Column                           Non-Null Count  Dtype 
---  ------                           --------------  ----- 
 0   PerformanceID                    6709 non-null   object
 1   EmployeeID                       6709 non-null   object
 2   ReviewDate                       6709 non-null   object
 3   EnvironmentSatisfaction          6709 non-null   int64 
 4   JobSatisfaction                  6709 non-null   int64 
 5   RelationshipSatisfaction         6709 non-null   int64 
 6   TrainingOpportunitiesWithinYear  6709 non-null   int64 
 7   TrainingOpportunitiesTaken       6709 non-null   int64 
 8   WorkLifeBalance                  6709 non-null   int64 
 9   SelfRating                       6709 non-null   int64 
 10  ManagerRating                    6709 non-null   int64 
dtypes: int64(8), object(3)
memory usage: 576.7+ KB


Defined DataFrame for PerformanceRating.csv, and checked for null values. 

In [17]:
emp_df = pd.read_csv("Employee.csv")
emp_df.info()
emp_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 23 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   EmployeeID               1470 non-null   object
 1   FirstName                1470 non-null   object
 2   LastName                 1470 non-null   object
 3   Gender                   1470 non-null   object
 4   Age                      1470 non-null   int64 
 5   BusinessTravel           1470 non-null   object
 6   Department               1470 non-null   object
 7   DistanceFromHome (KM)    1470 non-null   int64 
 8   State                    1470 non-null   object
 9   Ethnicity                1470 non-null   object
 10  Education                1470 non-null   int64 
 11  EducationField           1470 non-null   object
 12  JobRole                  1470 non-null   object
 13  MaritalStatus            1470 non-null   object
 14  Salary                   1470 non-null  

EmployeeID   FirstName LastName      Gender  Age BusinessTravel  \
0  3012-1A41    Leonelle    Simco      Female   30    Some Travel   
1  CBCB-9C9D     Leonerd    Aland        Male   38    Some Travel   
2  95D7-1CE9       Ahmed    Sykes        Male   43    Some Travel   
3  47A0-559B  Ermentrude   Berrie  Non-Binary   39    Some Travel   
4  42CC-040A       Stace   Savege      Female   29    Some Travel   

        Department  DistanceFromHome (KM) State                Ethnicity  ...  \
0            Sales                     27    IL                    White  ...   
1            Sales                     23    CA                    White  ...   
2  Human Resources                     29    CA  Asian or Asian American  ...   
3       Technology                     12    IL                    White  ...   
4  Human Resources                     29    CA                    White  ...   

   MaritalStatus  Salary StockOptionLevel OverTime    HireDate  Attrition  \
0       Divorced  102059                1       No  2012-01-03         No   
1         Single  157718                0      Yes  2012-01-04         No   
2        Married  309964                1       No  2012-01-04         No   
3        Married  293132                0       No  2012-01-05         No   
4         Single   49606                0       No  2012-01-05        Yes   

  YearsAtCompany YearsInMostRecentRole YearsSinceLastPromotion  \
0             10                     4                       9   
1             10                     6                      10   
2             10                     6                      10   
3             10                    10                      10   
4              6                     1                       1   

   YearsWithCurrManager  
0                     7  
1                     0  
2                     8  
3                     0  
4                     6  

[5 rows x 23 columns]

Merging DataFrames Employee.csv and PerformanceRate.csv

In [18]:
merged_df = pr_df.merge(emp_df, how='outer', on='EmployeeID', left_on=None, right_on=None, left_index=False, right_index=False, sort=False, suffixes=('_x', '_y'), copy=None, indicator=False, validate=None)

In [24]:
merged_df.info()
merged_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6899 entries, 0 to 6898
Data columns (total 33 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   PerformanceID                    6709 non-null   object 
 1   EmployeeID                       6899 non-null   object 
 2   ReviewDate                       6709 non-null   object 
 3   EnvironmentSatisfaction          6709 non-null   float64
 4   JobSatisfaction                  6709 non-null   float64
 5   RelationshipSatisfaction         6709 non-null   float64
 6   TrainingOpportunitiesWithinYear  6709 non-null   float64
 7   TrainingOpportunitiesTaken       6709 non-null   float64
 8   WorkLifeBalance                  6709 non-null   float64
 9   SelfRating                       6709 non-null   float64
 10  ManagerRating                    6709 non-null   float64
 11  FirstName                        6899 non-null   object 
 12  LastName            

PerformanceID EmployeeID ReviewDate  EnvironmentSatisfaction  \
0           NaN  001A-8F88        NaN                      NaN   
1        PR4067  005C-E0FB  6/17/2020                      3.0   
2        PR5070  005C-E0FB  6/17/2021                      4.0   
3        PR6165  005C-E0FB  6/17/2022                      3.0   
4        PR1165  00A3-2445  6/19/2016                      4.0   

   JobSatisfaction  RelationshipSatisfaction  TrainingOpportunitiesWithinYear  \
0              NaN                       NaN                              NaN   
1              3.0                       2.0                              1.0   
2              4.0                       5.0                              1.0   
3              4.0                       5.0                              3.0   
4              5.0                       4.0                              2.0   

   TrainingOpportunitiesTaken  WorkLifeBalance  SelfRating  ...  \
0                         NaN              NaN         NaN  ...   
1                         2.0              2.0         4.0  ...   
2                         1.0              5.0         3.0  ...   
3                         0.0              4.0         4.0  ...   
4                         2.0              5.0         4.0  ...   

   MaritalStatus  Salary StockOptionLevel OverTime    HireDate Attrition  \
0        Married   27763                0       No  2021-09-05        No   
1        Married   56155                1       No  2017-08-26        No   
2        Married   56155                1       No  2017-08-26        No   
3        Married   56155                1       No  2017-08-26        No   
4        Married  126238                0       No  2012-03-08        No   

  YearsAtCompany  YearsInMostRecentRole YearsSinceLastPromotion  \
0              1                      0                       1   
1              5                      2                       2   
2              5                      2                       2   
3              5                      2                       2   
4             10                      3                       6   

  YearsWithCurrManager  
0                    0  
1                    0  
2                    0  
3                    0  
4                    6  

[5 rows x 33 columns]

Count the number of null values in merged dataframe.

In [20]:
NullCells = merged_df["PerformanceID"].isnull().sum()
print(NullCells)

190


Out of 1470 employees, 190 don't have performance data, and 1280 that do.

Chart scatter plot with years since l;ast promotion vs Attrtion.

In [23]:
# Create the scatter plot
plt.figure(figsize=(10, 6))
plt.scatter(data['X-axis'], data['Y-axis'], color='blue', alpha=0.6)
plt.title('Scatter Plot')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')
plt.grid()

# Show the plot in Streamlit
st.pyplot(plt)

DeltaGenerator()